Library/Homebrew/download_strategy.rb

```diff
@@ -948,6 +1015,8 @@ def self.detect(url, strategy = nil)
 
   def self.detect_from_url(url)
     case url
+    when %r{^https?://github\.com/[\w-]+/[\w.-]+$}
```

Comment:
`%r{^https?://github\.com/[^/]+/[^/]+\.git$}`

Noted that you should only match GitDownloadStrategy, therefore the trailing `.git` cannot be omitted.